plugins/out_es/es.h

```diff
@@ -31,24 +34,24 @@
 #define FLB_ES_DEFAULT_TAG_KEY    "flb-key"
 #define FLB_ES_DEFAULT_HTTP_MAX   "512k"
 #define FLB_ES_DEFAULT_HTTPS_PORT 443
-#define FLB_ES_WRITE_OP_INDEX     "index"
-#define FLB_ES_WRITE_OP_CREATE    "create"
-#define FLB_ES_WRITE_OP_UPDATE    "update"
-#define FLB_ES_WRITE_OP_UPSERT    "upsert"
 
-struct flb_elasticsearch {
+struct flb_elasticsearch_config {
```

Comment:
The whole idea / implementation and naming of new types follows Upstream support in Forward output plugin.